rubicon/objc/api.py

```diff
@@ -393,6 +393,9 @@ def __init__(self, vartype=objc_id, weak=False):
         # Check here if we need a weak or strong ivar.
         self._ivar_weak = self.weak and not self._is_py_object
 
+        if self.weak and not (self._is_py_object or self._is_objc_object):
+            raise TypeError("Weak properties are only supported for Objective-C or Python object types")
```

Comment:
It might be useful to include the incorrect `vartype` in the error message. Most `TypeError`s in Python itself do this, and it makes debugging a little bit easier.

Reply:
Done.